integrated_channels/sap_success_factors/client.py

```diff
@@ -86,28 +86,33 @@ def __init__(self, enterprise_configuration):
         """
         super(SAPSuccessFactorsAPIClient, self).__init__(enterprise_configuration)
         self.global_sap_config = apps.get_model('sap_success_factors', 'SAPSuccessFactorsGlobalConfiguration').current()
-        self._create_session()
+        self.session = None
+        self.expires_at = None
 
     def _create_session(self):
         """
         Instantiate a new session object for use in connecting with SAP SuccessFactors
         """
-        session = requests.Session()
-        session.timeout = self.SESSION_TIMEOUT
-
-        oauth_access_token, expires_at = SAPSuccessFactorsAPIClient.get_oauth_access_token(
-            self.enterprise_configuration.sapsf_base_url,
-            self.enterprise_configuration.key,
-            self.enterprise_configuration.secret,
-            self.enterprise_configuration.sapsf_company_id,
-            self.enterprise_configuration.sapsf_user_id,
-            self.enterprise_configuration.user_type
-        )
+        now = datetime.datetime.utcnow()
+        if self.session is None or self.expires_at is None or now >= self.expires_at:
```

Comment:
this statement can be simplified like this
```
if self.expires_at is None or now >= self.expires_at:
```

Reply:
actually I added the same condition to make code consistent with another _create_session method that is in Degreed client.
https://github.com/edx/edx-enterprise/blob/master/integrated_channels/degreed/client.py#L190